src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,730 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+
+        // Road at the intersection the main road leads onto which the sliproad arrives onto
+        const auto crossing_road = [&] {
+            if (is_left_sliproad_turn)
+                return main_road_intersection->intersection.getLeftmostRoad();
+
+            if (is_right_sliproad_turn)
+                return main_road_intersection->intersection.getRightmostRoad();
+
+            BOOST_ASSERT_MSG(false, "Sliproad is neither a left nor right of obvious main road");
+            return main_road_intersection->intersection.getLeftmostRoad();
+        }();
+
+        const auto &crossing_road_data = node_based_graph.GetEdgeData(crossing_road.eid);
+
+        // The crossing road at the main road's intersection must not be incoming-only
+        if (crossing_road_data.reversed)
+        {
+            continue;
+        }
+
+        // Discard service and other low priority roads - never Sliproad candidate
+        if (sliproad_data.road_classification.IsLowPriorityRoadClass())
+        {
+            continue;
+        }
+
+        // Incoming-only can never be a Sliproad
+        if (sliproad_data.reversed)
         {
-            EdgeID candidate_in = road.eid;
-            const auto target_intersection = [&](NodeID node) {
-                auto intersection = intersection_generator(node, candidate_in);
-                // skip over traffic lights
-                if (intersection.size() == 2)
+            continue;
+        }
+
+        // This is what we know so far:
+        //
+        //       .
+        //       e
+        //       .
+        //       .
+        // a ... b .... c .   < `main_road_intersection` is intersection at `c`
+        //       `      .
+        //         `    .
+        //           `  .
+        //              d     < `target_intersection` is intersection at `d`
+        //              .       `sliproad_edge_target` is node `d`
+        //              e
+        //
+        //
+        //          ^ `sliproad` is `bd`
+        //       ^ `intersection` is intersection at `b`
+
+        if (!is_potential_link(sliproad))
+        {
+            continue;
+        }
+
+        // b -> d edge id
+        auto sliproad_edge = sliproad.eid;
+
+        // TODO: remove debugging
+        d = node_based_graph.GetTarget(sliproad_edge);
+
+        const auto target_intersection = [&](NodeID node) {
+            auto intersection = intersection_generator(node, sliproad_edge);
+            if (intersection.isTrafficSignalOrBarrier())
+            {
+                node = node_based_graph.GetTarget(sliproad_edge);
+                sliproad_edge = intersection[1].eid;
+                intersection = intersection_generator(node, sliproad_edge);
+            }
+            return intersection;
+        }(intersection_node_id);
+
+        // Constrain the Sliproad's target to sliproad, outgoing, incoming from main intersection
+        if (target_intersection.size() != 3)
```

Comment:
looks like a prime candidate to check early (node_based_graph.getOutDegree(target_node) == 3)